girder_worker/app.py

```diff
@@ -109,35 +126,54 @@ def girder_before_task_publish(sender=None, body=None, exchange=None,
             job_model = ModelImporter.model('job', 'jobs')
 
             user = headers.pop('girder_user', getCurrentUser())
-            token = headers.pop('girder_token', None)
-
             task_args, task_kwargs = body[0], body[1]
 
             job = job_model.createJob(
-                **{'title': headers.get('girder_job_title', Task._girder_job_title),
-                   'type': headers.get('girder_job_type', Task._girder_job_type),
-                   'handler': headers.get('girder_job_handler', Task._girder_job_handler),
-                   'public': headers.get('girder_job_public', Task._girder_job_public),
+                **{'title': headers.pop('girder_job_title', Task._girder_job_title),
+                   'type': headers.pop('girder_job_type', Task._girder_job_type),
+                   'handler': headers.pop('girder_job_handler', Task._girder_job_handler),
+                   'public': headers.pop('girder_job_public', Task._girder_job_public),
                    'user': user,
                    'args': task_args,
                    'kwargs': task_kwargs,
                    'otherFields': dict(celeryTaskId=headers['id'],
-                                       **headers.get('girder_job_other_fields',
+                                       **headers.pop('girder_job_other_fields',
```

Comment:
I'll admit that the code here is a little confusing,  but it is in service of a readable clean API for developers:

```python
fibonacci.delay(20,  girder_client_token="....")
```
vs.

```python
fibonacci.delay(20,  girder_headers={ "girder_client_token": "...." })
```

What's more this makes the distinction between kwargs that are used to control behavior in the ```before_task_publish``` signal vs options that are passed through to the message headers.  This means you would need a dict for ```girder_headers``` and a dict for ```girder_options```. This places unnecessary burden on the API user to know the difference between these things *or* means you've got to parse it out for them inside Task class. 

Each option that is possible has to be handled inside the Task code anyways to provide sensible defaults (in the case of kwargs being passed headers) or to provide functionality (in the case of kwargs effecting signal functionality).